crypto/fipsmodule/service_indicator/service_indicator_test.cc

```diff
@@ -2044,7 +2044,7 @@ TEST(ServiceIndicatorTest, RSAKeyGen) {
   ASSERT_TRUE(ctx);
 
   if (kEVPKeyGenShouldCallFIPSFunctions) {
-    // Test unapproved key sizes of RSA.
+    // Test various unapproved key sizes of RSA.
     for (const size_t bits : {512, 1024, 3071, 4095}) {
```

Comment:
Why 3071 and 4095 are still not approved after you discovered they're rounded down to be divisible by 128?

Reply:
Good question :)

That constraint aspect is enforced all the down in the private `rsa_generate_key_impl` function, which is what is ultimately invoked through either the `RSA_generate_key_ex` and `RSA_generate_key_fips` code paths. Since `RSA_generate_key_fips` always enforced the FIPS constraints upfront or failed the function I kept that same check here where it checks the requested bits upfront (>= 2048 and bits % 2 == 0). Thus is why its expected to be reject outright.

Reply:
Though I wonder if we should just enforce the `bits % 128 == 0` at this level which will also satisfy the divisible by two FIPS requirement. It ensures that the requested bits to the FIPS function will always return a valid FIPS key of the exacted requested length or error. Otherwise it’s still possible to call the FIPS versions, get a valid FIPS key divisible by two but not necessarily the exact bit length requested because it wasn’t divisible by 128.